dots/vimrc

```diff
@@ -161,43 +166,59 @@ let g:coffee_compile_vert=1
 nnoremap j gj
 nnoremap k gk
 
-" Visually select the text that was last edited/pasted
-nnoremap gV `[v`]
-
 " Move between splits
 noremap <C-h> <C-w>h
 noremap <C-j> <C-w>j
 noremap <C-k> <C-w>k
 noremap <C-l> <C-w>l
 
+" NERDTree in a buffer (like netrw)
+nnoremap <silent>- :silent edit %:p:h<CR>
+nnoremap <silent>_ :silent edit .<CR>
+
+" Override jumplist commands
+nnoremap <silent><C-i> :BufSurfBack<CR>
+nnoremap <silent><C-o> :BufSurfForward<CR>
+
 " Function keys
 let g:ctrlp_map='<F1>'
-noremap <F2> :NERDTreeToggle<cr>
-noremap <F3> :CtrlPBuffer<cr>
-nmap <F4> <Plug>(FerretAck)
-nmap <F5> <Plug>(FerretAckWord)
-vnoremap <F5> y:Ack -Q <C-r>=fnameescape(@")<cr><cr>
+noremap <F2> :NERDTreeToggle<CR>
+noremap <F3> :CtrlPBuffer<CR>
+nmap <F4> <Plug>CtrlSFQuickfixPrompt
+vmap <F4> <Plug>CtrlSFQuickfixVwordExec
+nmap <F5> <Plug>CtrlSFPrompt
+vmap <F5> <Plug>CtrlSFVwordExec
 noremap <F10> :vertical wincmd f<CR>
-noremap <F11> :BufSurfBack<cr>
-noremap <F12> :BufSurfForward<cr>
+noremap <F11> :BufSurfBack<CR>
+noremap <F12> :BufSurfForward<CR>
+
+" The `g` commands (testing)
+nnoremap <silent>gF :vertical wincmd f<CR>
+nnoremap <silent>gL :CtrlPBuffer<CR>
+nnoremap <silent>gl :CtrlP<CR>
+nnoremap gS :Ag<space>
+nmap gs <Plug>CtrlSFQuickfixPrompt
+vmap gs <Plug>CtrlSFQuickfixVwordExec
+nmap gz <Plug>CtrlSFPrompt
+vmap gz <Plug>CtrlSFVwordExec
+nnoremap <silent>gy :NERDTreeToggle<CR>
 
```

Comment:
I associate `g` with `go`. So `gf` is the native way to jump to file (`go file`) under the cursor, `gF` is our shortcut that does the same except in a vertical split. For `gl` and `gL`, all I could come up with is `go list` which fires up `CtrlP` or `CtrlPBuffer` respectively. The search in file results are mapped to `gs` or `go search` in both normal and visual mode. And I use `gz` as `go zearch` for the search and replace feature of CtrlSF. Saying the term `zearch` in your head is very `R` heavy so it kinda works? And the the `Y` in `gy` looks like a Tree maybe even a NERDTree.

For more great `g` commands type `help g`.